sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/ChallengeBasedAuthenticationPolicyTests.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Diagnostics;
+using Azure.Core.Pipeline;
+using Azure.Core.Testing;
+using Azure.Security.KeyVault.Tests;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Secrets.Tests
+{
+    public class ChallengeBasedAuthenticationPolicyTests
+    {
+        private const string TenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+        private const string AccessToken = "NzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3"; // base64(TenantId)
+        private const string VaultHost = "test.vault.azure.net";
+        private static readonly Uri s_vaultUri = new Uri("https://" + VaultHost);
+
+        [Test]
+        public async Task SingleRequest()
+        {
+            MockTransport transport = new MockTransportBuilder().Build();
+            SecretClientOptions options = new SecretClientOptions
+            {
+                Transport = transport,
+                Diagnostics =
+                {
+                    LoggedHeaderNames = { "*" },
+                    IsLoggingContentEnabled = true,
+                },
+            };
+
+            using AzureEventSourceListener logger = AzureEventSourceListener.CreateTraceLogger(EventLevel.Verbose);
+            SecretClient client = new SecretClient(s_vaultUri, new MockCredential(transport), options);
+
+            KeyVaultSecret secret = await client.GetSecretAsync("test-secret").ConfigureAwait(false);
+            Assert.AreEqual("secret-value", secret.Value);
+        }
+
+        [TestCase(10, 0, 0)]
+        [TestCase(10, 0, 500)]
+        public async Task MultipleRequests(int numberOfRequests, int minDelay, int maxDelay)
```

Comment:
Does this consistently repro the bug?

Reply:
It should since it basically replicates the repro app I wrote, but I can effectively back out my fix and make sure.

Reply:
Yes, let's see if it reproes the issue.

Reply:
After a few tweaks, I was able to get this to fail about 2/5 of the time on my laptop with the older runtime code.